### DISCUSSION TITLE
Contributing

### DISCUSSION THREAD

**Discussion by i-interprete-on-the-fly** [kdheepak/taskwarrior-tui#536] [General]
Hey,
I would like to contribute to several issues in this project in order to both increase my rust skills and also enhance one of my productity tools. I see that you are currently planning a major refactor (#349) so I'm not sure if I should wait until after that is finished. Can you share what the progress and timeline (if existing) on that is?

Thank you

**Comment by kdheepak** [comment]
Hi! Thanks for reaching out and offering to contribute! It's exciting to hear you want to help.

The refactor is still ongoing but I don't know if I have a concrete timeline yet (primarily because of my lack of time to work on it). I'm hoping some time will come up soon for me as I finish up various tasks on my todo list. 

I'm also learning new patterns and features in Ratatui / Rust as I explore different TUIs built using Ratatui and Rust. I'm also contributing to [Ratatui](https://github.com/ratatui-org/ratatui) as well (if you'd like to contribute there that'd be awesome too!). So I'm hedging on the refactor to wait for some features in Ratatui to land and some ideas to be fleshed out.

But I'm happy to merge smaller features or fix bugs at this point. Are there issues you want to see addressed?

**Comment by i-interprete-on-the-fly** [comment]
Hey! Alright, got you. So until that has made some progress I'll just try to tackle some smaller things bugging me. I have a small list and did open one or two issues, too.
Thanks!
